src/estimation/include/iDynTree/Estimation/AttitudeEstimator.h

```diff
@@ -0,0 +1,142 @@
+     *
+     * The aim is to implement different attitude estimators as a block that takes IMU measurements
+     * as inputs and gives attitude estimates as outputs. This way the underlying implementation is abstracted
+     * and the user only has to set a few parameters and run the estimator.
+     *
+     * The general procedure to use the estimators would be,
+     * - instantiate the filter,
+     * - set initial internal state
+     * - in a loop,
+     *     - update the filter with measurements
+     *     - propagate the states
+     *
+     * However, additional methods to set and get parameters for the filter might be available with respect to the filters.
+     */
+    class IAttitudeEstimator
+    {
+    public:
+        virtual ~IAttitudeEstimator();
+
+        /**
+         * @brief Update the filter with accelerometer and gyroscope measurements
+         *
+         * @param[in] linAccMeas left trivialized 3D vector of linear proper sensor acceleration measuements
+         * @param[in] gyroMeas left trivialized 3D vector of angular velocity measurements
+         *
+         * @note left trivialized angular velocity means the angular velocity of body frame B with respect to an inertial fram A, expressed in frame B
+         * @return true/false if successful/not
+         */
+        virtual bool updateFilterWithMeasurements(const iDynTree::LinearAccelerometerMeasurements& linAccMeas,
+                                          const iDynTree::GyroscopeMeasurements& gyroMeas) = 0;
+
+        /**
+         * @brief Update the filter with accelerometer, gyroscope and magnetometer measurements
+         *
+         * @param[in] linAccMeas left trivialized 3D vector of linear proper sensor acceleration measuements
+         * @param[in] gyroMeas left trivialized 3D vector of angular velocity measurements
+         * @param[in] magMeas left trivialized 3D vector of magnetometer measurements
+         *
+         * @note left trivialized angular velocity means the angular velocity of body frame B with respect to an inertial fram A, expressed in frame B
+         *
+         * @return true/false if successful/not
+         */
+        virtual bool updateFilterWithMeasurements(const iDynTree::LinearAccelerometerMeasurements& linAccMeas,
+                                                  const iDynTree::GyroscopeMeasurements& gyroMeas,
+                                                  const iDynTree::MagnetometerMeasurements& magMeas) = 0;
+
+        /**
+         * @brief Propagate the states and associated uncertainties through properly defined propagation functions
+         * The underlying implementation depends on the type of filter being implemented.
+         *
+         * @return true/false if successful/not
+         */
+        virtual bool propagateStates() = 0;
+
+        /**
+         * @brief Get orientation of the body with respect to inertial frame, in rotation matrix form
```

Comment:
I think if you also write mathematical expression  such as `\f$ {}^A R_B \f$` it would be beneficial for the users (not only here about in all the class).

Reply:
Doesn't the documentation of virtual functions get carried over to the child classes?

Reply:
> Doesn't the documentation of virtual functions get carried over to the child classes?

I am not sure how this is related to my comment.

Reply:
>  (not only here but in all the class).
For all the classes (derived), whatever is being documented here is carried over, if I'm right?. In that sense.

Reply:
Sorry, I clarified the original comment: I meant in all the methods of this class.